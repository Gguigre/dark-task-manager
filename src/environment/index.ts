export const env = {
  ENV: 'local',
  featureFlags: {
    areCheatCodesEnabled: true,
    isCodePushEnabled: false,
    isCodePushManual: false,
  },
};
