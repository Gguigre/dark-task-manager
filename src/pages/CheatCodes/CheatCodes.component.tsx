import React from 'react';
import { View } from 'react-native';
import { NavigationActions, NavigationInjectedProps } from 'react-navigation';
import { CodePushButton } from './components/CodePushButton';
import { CrashTestButton } from './components/CrashTestButton';
import { ChangeLanguage } from './components/ChangeLanguage';
import { env } from '../../environment';

export const CheatCodes = (props: NavigationInjectedProps<{}>) => (
  <View>
    <CrashTestButton />
    <ChangeLanguage
      onLanguageChange={() =>
        // @ts-ignore
        props.navigation.reset([NavigationActions.navigate({ routeName: 'Login' })], 0)
      }
    />
    {env.featureFlags.isCodePushEnabled && <CodePushButton />}
  </View>
);
