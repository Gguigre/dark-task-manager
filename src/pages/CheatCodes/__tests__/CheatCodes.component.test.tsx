import React from 'react';
import { render, act } from 'react-native-testing-library';
import { CheatCodes } from '../CheatCodes.component';
import { env } from '../../../environment';

jest.mock('../../../environment', () => ({
  env: {
    featureFlags: {
      isCodePushEnabled: true,
    },
  },
}));

describe('CheatCodes component', () => {
  const navigation = {
    reset: jest.fn(),
  } as any; // eslint-disable-line @typescript-eslint/no-explicit-any

  it('should have a button to go to the Login', async () => {
    const cheatCodes = render(<CheatCodes navigation={navigation} />);
    expect(cheatCodes).toMatchSnapshot();
  });

  it('should have a button to go to the Login', async () => {
    const cheatCodes = render(<CheatCodes navigation={navigation} />);

    cheatCodes.getByText('Check update');
    const Picker = cheatCodes.getByTestId('changeLanguagePicker');
    act(() => {
      Picker.props.onValueChange('EN');
    });

    expect(navigation.reset).toHaveBeenCalled();
  });

  it('should have a button to go to the Login', async () => {
    env.featureFlags.isCodePushEnabled = false;
    const cheatCodes = render(<CheatCodes navigation={navigation} />);
    expect(() => cheatCodes.getByText('Check update')).toThrowError();
  });
});
