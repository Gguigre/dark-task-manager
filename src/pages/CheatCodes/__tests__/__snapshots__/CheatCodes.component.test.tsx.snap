// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheatCodes component should have a button to go to the Login 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Crash the app
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 50,
        "width": 100,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "FR",
            "textColor": undefined,
            "value": "fr",
          },
          Object {
            "label": "EN",
            "textColor": undefined,
            "value": "en",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="changeLanguagePicker"
    />
  </View>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      Check update
    </Text>
  </View>
</View>
`;
